src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -447,9 +432,7 @@ private function createChoiceListView(ChoiceListInterface $choiceList, array $op
         // If no explicit grouping information is given, use the structural
         // information from the "choices" option for creating groups
         if (!$options['group_by'] && $options['choices']) {
-            $options['group_by'] = !$options['choices_as_values']
-                ? self::flipRecursive($options['choices'])
-                : $options['choices'];
+            $options['group_by'] = $options['choices'];
```

Comment:
@xabbuh Thanks for the review.

`choices_as_values` is [normalized](https://github.com/symfony/symfony/blob/3.0/src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php#L280) to `true` since 3.0 and deprecated in 3.1.

This change is safe.